src/MLJMultivariateStatsInterface.jl

```diff
@@ -150,4 +150,1078 @@ metadata_pkg.(
+- `alg::Symbol=:fastica`: The algorithm to use (only `:fastica` is supported at the moment).
+- `fun::Symbol=:tanh`: The approximate neg-entropy function, one of `:tanh`, `:gaus`.
+- `do_whiten::Bool=true`: Whether or not to perform pre-whitening.
+- `maxiter::Int=100`: The maximum number of iterations.
+- `tol::Real=1e-6`: The convergence tolerance for change in the unmixing matrix W.
+- `mean::Union{Nothing, Real, Vector{Float64}}=nothing`: mean to use, if nothing (default)
+   centering is computed and applied, if zero, no centering; otherwise a vector of means can
+   be passed.
+- `winit::Union{Nothing,Matrix{<:Real}}=nothing`: Initial guess for the unmixing matrix
+   `W`: either an empty matrix (for random initilization of `W`), a matrix of size `m × k`
+   (if `do_whiten` is true), or a matrix of size `m × k`. Here `m` is the number
+   of components (columns) of the input.
+
+# Operations
+
+- `transform(mach, Xnew)`: Return the component-separated version of input
+   `Xnew`, which should have the same scitype as `X` above.
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+# TODO: Now that this is fixed, document
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `indim`: Dimension (number of columns/components) of the training
+   data and new data to be transformed.
+- `outdim`: Dimension of transformed data (number of separated components).
+- `mean`: The mean vector, which has length `indim`.
+
+# Examples
+
+```
+using MLJ
+using LinearAlgebra
+
+ICA = @load ICA pkg=MultivariateStats
+
+X, y = @load_iris
+
+model = ICA(k = 2, tol=0.1)
+mach = machine(model, X) |> fit!
+
+Xproj = transform(mach, X)
+```
+
+See also
+[`PCA`](@ref), [`KernelPCA`](@ref), [`FactorAnalysis`](@ref), [`PPCA`](@ref)
+"""
+ICA
+"""
+$(MMI.doc_header(LDA))
+
+`LDA`: Multiclass linear discriminant analysis. The algorithm learns a
+projection matrix `P` that projects a feature matrix `Xtrain` onto a lower dimensional
+space of dimension `out_dim` such that the trace of the transformed between-class
+scatter matrix(`Pᵀ*Sb*P`) is maximized relative to the trace of the transformed
+within-class scatter matrix (`Pᵀ*Sw*P`).The projection matrix is scaled such that
+`Pᵀ*Sw*P=I` or `Pᵀ*Σw*P=I`(where `Σw` is the within-class covariance matrix) .
+Predicted class posterior probability for feature matrix `Xtest` are derived by
+applying a softmax transformationto a matrix `Pr`, such that  rowᵢ of `Pr` contains
+computed distances(based on a distance metric) in the transformed space of rowᵢ in
+`Xtest` to the centroid of each class.
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+
+    mach = machine(model, X)
+
+Where
+
+- `X`: is any table of input features (eg, a `DataFrame`) whose columns
+  are of scitype `Continuous`; check the scitypes with `schema(X)`
+- `y`: is the target, which can be any `AbstractVector` whose element
+  scitype is `<:OrderedFactor(2)` or `<:Multiclass(2)`; check the scitype
```

Comment:
This does not agree with the LDA `metadata_model` declaration. Please check these very carefully. FYI `Finite` is the super type of `OrderedFactor` and `Multiclass` but as users a maybe less familiar with it, I generally unwrap "`Finite`" as "`OrderedFactor` or `Multiclass`" as here. 

```suggestion
  scitype is `OrderedFactor` or `Multiclass`; check the scitype
```